WcaOnRails/app/models/user.rb

```diff
@@ -327,6 +327,34 @@ def avatar_requires_wca_id
     end
   end
 
+  after_save :changed_delegate_status
+  private def changed_delegate_status
```

Comment:
As a general rule, we prefer to not have models send emails. Instead, we prefer to send emails explicitly from our controllers. See https://github.com/jfly/worldcubeassociation.org/blob/efcd848621418fd54dd6d7c8be42faf2f2dd1511/WcaOnRails/app/controllers/delegate_reports_controller.rb#L41 for an example of where we send the delegate report emails.

In this case, I think you'd want to move this to the update method of `users_controller`.

Reply:
At first I tried to make this in the controller, but I was finding it difficult. After I saw an example on the model, I decided to try it this way and it was much easier. I will go back with my first idea and do it in the controller.